### ISSUE TITLE
Uncertainty based exploration

### ISSUE THREAD

**Issue by raharth** [raharth/PyTorch_Lib#59]
Link exploration rate to uncertainty of the model.

This can be e.g. done by using a softmax with a temp parameter, that is proportional to the uncertainty.

The main question them becomes how to choose the uncertainty -> temp mapping. If we want to have a general approach this probably has to be a rescaled variable, since we otherwise depend on the reward size.

Maybe use a normal distribution of the uncertainty with an exponential function. By this we would have a state it is average certain about at the predefined temperature (since x^0 = 1)

**Comment by raharth** [comment]
Run experiment 170
